arbiter-core/src/environment.rs

```diff
@@ -287,6 +329,58 @@ pub enum EnvironmentError {
     NotRandomlySampledBlockType,
 }
 
+/// The `EnvironmentBuilder` is a builder pattern for creating an [`Environment`].
+/// It allows for the configuration of the [`Environment`] before it is created.
+impl EnvironmentBuilder {
+    /// Creates a new `EnvironmentBuilder` with default settings.
+    /// By default, the `block_settings` and `gas_settings` are set to `UserControlled`.
+    pub fn new() -> Self {
+        Self {
+            label: None,
+            block_settings: BlockSettings::UserControlled,
+            gas_settings: GasSettings::UserControlled,
+        }
+    }
+
+    /// Sets the `block_settings` for the `EnvironmentBuilder`.
+    /// This determines how the block number and timestamp are controlled in the [`Environment`].
+    pub fn block_settings(mut self, block_settings: BlockSettings) -> Self {
+        self.block_settings = block_settings;
+        self
+    }
+
+    /// Sets the `gas_settings` for the `EnvironmentBuilder`.
+    /// This determines how the gas price is controlled in the [`Environment`].
+    pub fn gas_settings(mut self, gas_settings: GasSettings) -> Self {
+        self.gas_settings = gas_settings;
+        self
+    }
+
+    /// Sets the `label` for the `EnvironmentBuilder`.
+    /// This is an optional string that can be used to identify the [`Environment`].
+    pub fn label(mut self, label: String) -> Self {
+        self.label = Some(label);
+        self
+    }
```

Comment:
Can we do this instead?
```suggestion
    pub fn label(mut self, label: impl Into<String>) -> Self {
        self.label = Some(label.into());
        self
    }
```